src/dftbp/dftb/rshgamma.F90

```diff
@@ -0,0 +1,500 @@
+    real(dp) :: diag1(3), diag2(3), diag3(3)
+
+    !! Normal vectors of three planes of the cell
+    real(dp) :: n1(3), n2(3), n3(3)
+
+    !! Point of intersection in absolute coordinates
+    real(dp) :: intersect(3)
+
+    !! Distances of sphere origin to cell planes
+    real(dp) :: distances(3)
+
+    !! Origin dummy
+    real(dp) :: origin(3)
+
+    !! Auxiliary variable
+    integer :: ii
+
+    @:ASSERT(size(latVecs, dim=1) == 3)
+    @:ASSERT(size(latVecs, dim=2) == 3)
+
+    origin(:) = 0.0_dp
+
+    if (present(nK)) then
+      @:ASSERT(size(nK) == 3)
+      nK_(:) = nK
+    else
+      nK_(:) = 1
+    end if
+
+    do ii = 1, 3
+      bvkLatVecs(:, ii) = nK_ * latVecs(:, ii)
+    end do
+
+    diag1(:) = bvkLatVecs(:, 1) + bvkLatVecs(:, 2) + bvkLatVecs(:, 3)
+    diag2(:) = bvkLatVecs(:, 1) - bvkLatVecs(:, 2) + bvkLatVecs(:, 3)
+    diag3(:) = bvkLatVecs(:, 2) - bvkLatVecs(:, 1) + bvkLatVecs(:, 3)
+
+    intersect = getDiagIntersectionPoint(bvkLatVecs(:, 2), bvkLatVecs(:, 2) + diag2, origin,&
+        & origin + diag1)
+
+    ! calculate normal vectors of three planes
+    n1 = cross3(bvkLatVecs(:, 1), bvkLatVecs(:, 2))
+    n1 = n1 / norm2(n1)
+
+    n2 = cross3(bvkLatVecs(:, 3), bvkLatVecs(:, 1))
+    n2 = n2 / norm2(n2)
+
+    n3 = cross3(bvkLatVecs(:, 2), bvkLatVecs(:, 3))
+    n3 = n3 / norm2(n3)
+
+    distances(1) = abs(dot_product(intersect, n1))
+    distances(2) = abs(dot_product(intersect, n2))
+    distances(3) = abs(dot_product(intersect, n3))
+
+    cutoff = minval(distances)
+
+  contains
+
+    !> Returns intersection point of two diagonals in a triclinic cell.
+    function getDiagIntersectionPoint(startDiag1, endDiag1, startDiag2, endDiag2)&
+        & result(intersect1)
+
+      !> Start and end point of first diagonal
+      real(dp), intent(in) :: startDiag1(:), endDiag1(:)
+
+      !> Start and end point of second diagonal
+      real(dp), intent(in) :: startDiag2(:), endDiag2(:)
+
+      !> Points of intersection in absolute coordinates
+      real(dp) :: intersect1(3), intersect2(3)
+
+      !! Deviation between to calculated intersection points
+      real(dp) :: deviation
+
+      !! Parameters of diagonal lines
+      real(dp) :: tt, ss
+
+      ! calculate parameters of diagonal lines
+      tt = ((startDiag2(1) - startDiag1(1)) * (startDiag2(2) - endDiag2(2))&
+          & - (startDiag2(2) - startDiag1(2)) * (startDiag2(1) - endDiag2(1)))&
+          & / ((startDiag2(2) - endDiag2(2)) * (endDiag1(1) - startDiag1(1))&
+          & - (startDiag2(1) - endDiag2(1)) * (endDiag1(2) - startDiag1(2)))
+      ss = ((endDiag1(2) - startDiag1(2)) * (startDiag2(1) - startDiag1(1))&
+          & - (endDiag1(1) - startDiag1(1)) * (startDiag2(2) - startDiag1(2)))&
+          & / ((endDiag1(2) - startDiag1(2)) * (startDiag2(1) - endDiag2(1))&
+          & - (endDiag1(1) - startDiag1(1)) * (startDiag2(2) - endDiag2(2)))
+
+      ! check if third eq. also satisfied (3 equations but 2 variables)
```

Comment:
thrid eq. of what? Add paper ref somewhere in this source.

Reply:
This is simple trigonometry really, I think I even over-complicated this implementation. One equation for each component (x,y,z). I will slightly adapt the comment(s).